compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -341,6 +344,30 @@ impl<'hir> LoweringContext<'_, 'hir> {
         cond: &Expr,
         then: &Block,
         else_opt: Option<&Expr>,
+    ) -> hir::ExprKind<'hir> {
+        macro_rules! rslt {
+            ($opt:expr) => {{
+                let then_expr = self.lower_block_expr(then);
+                hir::ExprKind::If(self.lower_expr(cond), self.arena.alloc(then_expr), $opt)
+            }};
+        }
+        if let Some(rslt) = else_opt {
+            if let ExprKind::Let(ref pat, ref scrutinee) = rslt.kind {
```

Comment:
Hasn't this already been checked for in `lower_expr_mut`?

Reply:
Hum... this slipped through the cracks. Thanks!